cmd/sniproxy/snipproxy.go

```diff
@@ -85,9 +90,30 @@ func (s *server) serveDNSConn(c nettype.ConnPacketConn) {
 	defer c.Close()
 	c.SetReadDeadline(time.Now().Add(5 * time.Second))
 	buf := make([]byte, 1500)
-	n, err := c.Read(buf)
-	log.Printf("got DNS packet: %q, %v", buf[:n], err)
-	// TODO: rest of the owl
+	_, err := c.Read(buf)
```

Comment:
you need to get `n` and then use it below in `msg.Unpack(buf[:n])`

Reply:
Done.